cmd/testpeers_internal_test.go

```diff
@@ -224,9 +199,15 @@ func TestPeersTest(t *testing.T) {
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			temp := t.TempDir()
+			_, err := p2p.NewSavedPrivKey(temp)
+			require.NoError(t, err)
+			cfg := test.config
+			cfg.DataDir = temp
+
 			var buf bytes.Buffer
 			ctx := context.Background()
-			err := runTestPeers(ctx, &buf, test.config)
+			err = runTestPeers(ctx, &buf, cfg)
 			if test.expectedErr != "" {
 				require.ErrorContains(t, err, test.expectedErr)
```

Comment:
It might be the case that errors are coming from external libraries and are wrapped, so I decided to keep it more generic. Currently there are mostly newly created errors, but don't want to force that for all test cases.

I plan to implement `natOpen` in my next PR, so not sure if necessary defining it now.